clients/spark/core/src/main/scala/io/treeverse/clients/ApiClient.scala

```diff
@@ -50,13 +56,24 @@ class ApiClient(apiUrl: String, accessKey: String, secretKey: String) {
     )
   }
 
+  def prepareGarbageCollectionCommits(
+      repoName: String,
+      previousRunID: String
+  ): GarbageCollectionPrepareResponse = {
+    retentionApi.prepareGarbageCollectionCommits(
+      repoName,
+      new GarbageCollectionPrepareRequest().previousRunId(previousRunID)
+    )
+  }
+
   def getMetaRangeURL(repoName: String, commitID: String): String = {
     val commit = commitsApi.getCommit(repoName, commitID)
     val metaRangeID = commit.getMetaRangeId
-
-    val metaRange = metadataApi.getMetaRange(repoName, metaRangeID)
-    val location = metaRange.getLocation
-    URI.create(getStorageNamespace(repoName) + "/" + location).normalize().toString
+    if (metaRangeID != "") {
+      val metaRange = metadataApi.getMetaRange(repoName, metaRangeID)
+      val location = metaRange.getLocation
+      URI.create(getStorageNamespace(repoName) + "/" + location).normalize().toString
```

Comment:
As you probably know, I am not a fan of of constructing URIs by hand and prefer `resolve`: that is specified by a standard.  The issue here is sensitivity to trailing and leading slashes.
(Not a blocker)